core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AddExchangesBelowPartialAggregationOverGroupIdRuleSet.java

```diff
@@ -104,25 +104,33 @@
             typeOf(ExchangeNode.class)
                     .with(scope().equalTo(REMOTE))
                     .with(source().matching(
-                            // PushPartialAggregationThroughExchange adds a projection. However, it can be removed if RemoveRedundantIdentityProjections is run in the mean-time.
-                            typeOf(ProjectNode.class).capturedAs(PROJECTION)
+                            typeOf(ProjectNode.class)
                                     .with(source().matching(
-                                            typeOf(AggregationNode.class).capturedAs(AGGREGATION)
-                                                    .with(step().equalTo(AggregationNode.Step.PARTIAL))
-                                                    .with(nonEmpty(groupingColumns()))
+                                            typeOf(ExchangeNode.class)
+                                                    .with(scope().equalTo(LOCAL))
```

Comment:
any Local exchange is fine?

Reply:
In my understanding this rule is trying to capture a specific shape of a plan that got change with introduction of an additional local exchange. Thus in this plane shape other types of local exchange shouldn't be expected. But it would be better to wait for @findepi to take a look to make sure my understanding is correct.

Reply:
@arhimondr sorry, i don't remember. it feels like i wrote this code like 5 years ago